chips/rp2040/src/pwm.rs

```diff
@@ -0,0 +1,1290 @@
+        self.stop_pwm_channel(channel_number)
+    }
+
+    /// Return the maximum value of the frequency in Hz
+    ///
+    /// ## Panics
+    ///
+    /// This method will panic if the dependencies are not resolved.
+    fn get_maximum_frequency_hz(&self) -> usize {
+        self.clocks
+            .unwrap_or_panic()
+            .get_frequency(clocks::Clock::System) as usize
+    }
+
+    /// Return an opaque value representing 100% duty cycle
+    fn get_maximum_duty_cycle(&self) -> usize {
+        u16::MAX as usize + 1
+    }
+}
+
+/// Helper structure to control a PWM pin
+pub struct PwmPin<'a> {
+    pwm_struct: &'a Pwm<'a>,
+    channel_number: ChannelNumber,
+    channel_pin: ChannelPin,
+}
+
+impl PwmPin<'_> {
+    /// Returns the PWM channel the pin belongs to
+    pub fn get_channel_number(&self) -> ChannelNumber {
+        self.channel_number
+    }
+
+    /// Returns the PWM pin the pin belongs to
+    pub fn get_channel_pin(&self) -> ChannelPin {
+        self.channel_pin
+    }
+
+    // See [Pwm::set_invert_polarity_a] and [Pwm::set_invert_polarity_b]
+    fn set_invert_polarity(&self, inv: bool) {
+        if self.channel_pin == ChannelPin::A {
+            self.pwm_struct
+                .set_invert_polarity_a(self.channel_number, inv);
+        } else {
+            self.pwm_struct
+                .set_invert_polarity_b(self.channel_number, inv);
+        }
+    }
+
+    // See [Pwm::set_compare_value_a] and [Pwm::set_compare_value_b]
+    fn set_compare_value(&self, compare_value: u16) {
+        if self.channel_pin == ChannelPin::A {
+            self.pwm_struct
+                .set_compare_value_a(self.channel_number, compare_value);
+        } else {
+            self.pwm_struct
+                .set_compare_value_b(self.channel_number, compare_value);
+        }
+    }
+}
+
+impl hil::pwm::PwmPin for PwmPin<'_> {
+    /// Same as Pwm::start
+    fn start(&self, frequency_hz: usize, duty_cycle: usize) -> Result<(), ErrorCode> {
+        self.pwm_struct.start_pwm_pin(
+            self.channel_number,
+            self.channel_pin,
+            frequency_hz,
+            duty_cycle,
+        )
+    }
+
+    /// Same as Pwm::stop
+    fn stop(&self) -> Result<(), ErrorCode> {
+        self.pwm_struct.stop_pwm_channel(self.channel_number)
+    }
+
+    /// Same as Pwm::get_maximum_frequency_hz
+    fn get_maximum_frequency_hz(&self) -> usize {
+        hil::pwm::Pwm::get_maximum_frequency_hz(self.pwm_struct)
+    }
+
+    /// Same as Pwm::get_maximum_duty_cycle
+    fn get_maximum_duty_cycle(&self) -> usize {
+        hil::pwm::Pwm::get_maximum_duty_cycle(self.pwm_struct)
+    }
+}
+
+/// Interrupt trait for interrupt handlers
+///
+/// One should implement this trait if they need to perform some action
+/// on PWM interrupts. A PWM interrupt is raised when a counter wraps if running
+/// in trailing-edge mode or when it reaches again the value 0 when running in dual
+/// slope mode.
+pub trait Interrupt {
```

Comment:
What would be a use case for this trait?

Reply:
Removed it since the HIL provides no support for interrupts.